src/main/java/io/reactivex/functions/ThrowablePermittingConsumer.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2016-present, RxJava Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.functions;
+
+/**
+ * A functional interface (callback) that accepts a single value and permits throwing of a
+ * {@link Throwable} from the {@link #accept(Object)} method.
+ *
+ * @param <T> the value type
+ */
+public interface ThrowablePermittingConsumer<T> {
```

Comment:
Maybe reuse Java's `UncaughtExceptionHandler` instead of adding new interface to RxJava?